engine/access/rest/middleware/logging_middleware.go

```diff
@@ -0,0 +1,56 @@
+package middleware
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+)
+
+// LoggingMiddleware creates a middleware which adds a logger interceptor to each request to log the request method, uri,
+// duration and response code
+func LoggingMiddleware(logger zerolog.Logger) mux.MiddlewareFunc {
+	return func(inner http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			// record star time
+			start := time.Now()
+			// modify the writer
+			respWriter := newResponseWriter(w)
+			// continue to the next handler
+			inner.ServeHTTP(respWriter, req)
```

Comment:
switched back the logging middleware to do the `doSomething` -> `call next handler` -> `doSomething` pattern since gorilla mux middlewares are all executed _before_ the handler. 
there were other ways of accomplishing this but they would be non-standard making the code a little complicated to read. I would rather stick with the Gorilla mux way of doing it.